packages/ux-capture/src/Zone.js

```diff
@@ -19,7 +39,36 @@ function Zone(props) {
 	this.measureName = this.props.name;
 
 	// Create a new `ExpectedMark` for each mark
-	this.marks = this.props.marks.map(markName => {
+	const configuredMarkNames = this.props.elements
+		? // new elements array on zone object
+		this.props.elements.map(element => element.marks).flat()
+		: // legacy with direct marks array on zone object
+		this.props.marks;
+
+	// mark names for elements already on the page after interactive transition
+	let elementsAlreadyOnThePage = [];
+
+	if (this.props.startMarkName === INTERACTIVE_TRANSITION_START_MARK_NAME &&
+		this.props.elements &&
+		Array.isArray(this.props.elements)
+	) {
+		elementsAlreadyOnThePage = this.props.elements.filter(element => {
+			const nodes = this.selectDOMNodes(element);
+
+			// if array or element list is returned, check if it has one or more entries
+			return (nodes && (typeof nodes.length === 'undefined' || nodes.length > 0));
+		});
```

Comment:
I think you could just make this a set of existing marks on the page? That way it would have a constant lookup time instead of the nested `find`s repeated for each "configured mark name" later. Something like this:
```
const existingMarks = new Set();

if (...) {
    this.props.elements.forEach((element) => {
        const nodes = this.selectDOMNodes(element);

        if (nodes && (typeof nodes.length === 'undefined' || nodes.length > 0)) {
            element.marks.forEach((mark) => existingMarks.add(mark));
        }
    });
}

const markNamesToExpect = configuredMarkNames.filter((mark) => !existingMarks.has(mark));
```

Reply:
I am not as concerned with time as I am with readability though as datasets are small, but set operations might be more readable too. Thank you for a suggestion.

I do want to create a list of elements to make it clear that we are not operating on mark level here, but on element level - I'll see what I can do about this.

Reply:
Now that I am thinking about it - using sets might also remove the bugs caused by marks being defined multiple times and all that.